pyramid_swagger/load_schema.py

```diff
@@ -44,100 +46,74 @@ def build_param_schema(schema, param_type):
-    Which we can then validate against a JSON object we construct from the
-    pyramid request.
-    """
-    matching_body_schemas = [
-        s
-        for s in schema['parameters']
-        if s['paramType'] == 'body'
-    ]
-    if matching_body_schemas:
-        schema = matching_body_schemas[0]
-        type_ref = extract_validatable_type(schema['type'], models_schema)
-        # Unpleasant, but we are forced to replace 'type' defns with proper
-        # jsonschema refs.
-        if '$ref' in type_ref:
-            del schema['type']
-            schema.update(type_ref)
-        return strip_swagger_markup(schema)
-    else:
-        return None
 
+def extract_body_schema(schema):
+    """Return the body parameter schema from an operation schema."""
+    matching_body_schemas = filter_params_by_type(schema, 'body')
+    # There can be only one body param
+    return matching_body_schemas[0] if matching_body_schemas else None
 
-# TODO: do this with jsonschema directly
-def strip_swagger_markup(schema):
-    """Turn a swagger URL parameter schema into a raw jsonschema.
 
-    Involves just removing various swagger-specific markup tags.
-    """
-    swagger_specific_keys = (
-        'paramType',
-        'name',
-    )
-    return dict(
-        (k, v)
-        for k, v in schema.items()
-        if k not in swagger_specific_keys
-    )
+def noop_validator(_validator, *args):
+    return
 
 
-def get_model_resolver(schema):
-    """
-    Get a RefResolver. RefResolver's will resolve "$ref:
-    ObjectType" entries in the schema, which are used to describe more complex
-    objects.
+def build_swagger_type_validator(models):
+    def swagger_type_validator(validator, ref, instance, schema):
+        func = _validators.ref if ref in models else _validators.type_draft4
+        return func(validator, ref, instance, schema)
+
+    return swagger_type_validator
+
 
-    :returns: The RefResolver for the schema's models.
-    :rtype: RefResolver
+def required_validator(validator, req, instance, schema):
+    """Swagger 1.2 expects `required` to be a bool in the Parameter object, but
+    a list of properties in a Model object.
     """
-    models = dict(
-        (k, strip_swagger_markup(v))
-        for k, v in schema.get('models', {}).items()
+    if schema.get('paramType') and req is True:
+        if not instance:
+            return [ValidationError("%s is required" % schema['name'])]
+    else:
+        return _validators.required_draft4(validator, req, instance, schema)
+
+
+def get_body_validator(models):
+    return validators.extend(
+        Draft4Validator,
+        {
+            'paramType': noop_validator,
+            'name': noop_validator,
+            'type': build_swagger_type_validator(models),
+            'required': required_validator,
+        }
     )
-    return RefResolver('', '', models)
+
+
+Swagger12ParamValidator = validators.extend(
+    Draft3Validator,
+    {
+        'paramType': noop_validator,
```

Comment:
I thought about it some more, and I think what this is really saying is `ignore` instead of `noop`.  Basically we have these fields in the swagger schema, and they don't actually represent anything about validation, they're for other parts of swagger, so for the purposes of validation we're ignoring them.

I'm going to change the name of this validator to `ignore` instead of `noop_validator`.